src/daemon/collectd.c

```diff
@@ -191,13 +191,15 @@ static int change_basedir(const char *orig_dir) {
     return -1;
   }
 
-  status = mkdir(dir, S_IRWXU | S_IRWXG | S_IRWXO);
-  if (status != 0) {
-    char errbuf[1024];
-    ERROR("change_basedir: mkdir (%s): %s", dir,
-          sstrerror(errno, errbuf, sizeof(errbuf)));
-    free(dir);
-    return -1;
+  if (nocreate == 0) {
```

Comment:
This tries the `chdir()` call a second time, which is not elegant. How about we extend the above logic to:

```c
else if (!create || (errno != ENOENT)) {
  //     ^^^^^^^^^^
  /* report error, return */
}
```

Reply:
That's better, I agree.